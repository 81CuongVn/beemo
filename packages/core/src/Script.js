/**
 * @copyright   2017, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 * @flow
 */

import { Module } from 'boost';

import type { Tool, Reporter } from 'boost';
import type Driver from './Driver';

export default class Script<To: Object = {}> extends Module<To> {
  /**
   * Run the script with the options object and Beemo tool instance.
   */
  run(options: Object, tool: Tool<Driver, Reporter<Object>>): Promise<*> {
    throw new Error(`${this.constructor.name} must define a run() method.`);
  }

  /**
   * Pass a configuration object to parse args with.
   */
  setup(): Object {
    return {};
  }
}
