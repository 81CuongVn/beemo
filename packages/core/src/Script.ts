/**
 * @copyright   2017, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 */

import { Module, Tool } from '@boost/core';
import { Options, Arguments } from 'yargs';

export default class Script<Opts = {}> extends Module<Opts> {
  /**
   * Define a configuration object to parse args with.
   */
  parse(): { [option: string]: Options } {
    return {};
  }

  /**
   * Run the script with the options object and Beemo tool instance.
   */
  async run(args: Arguments, tool: Tool): Promise<any> {
    throw new Error(`${this.constructor.name} must define an async run() method.`);
  }
}
