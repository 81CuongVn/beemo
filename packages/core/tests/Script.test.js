import Script from '../src/Script';

describe('Script', () => {
  let script;

  beforeEach(() => {
    script = new Script();
  });

  describe('parse()', () => {
    it('returns an empty object', () => {
      expect(script.parse()).toEqual({});
    });
  });

  describe('run()', () => {
    it('errors if not defined', () => {
      expect(() => {
        script.run();
      }).toThrowErrorMatchingSnapshot();
    });
  });
});
