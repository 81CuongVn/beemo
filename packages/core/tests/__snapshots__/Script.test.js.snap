// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Script run() errors if not defined 1`] = `"Script must define a run() method."`;
